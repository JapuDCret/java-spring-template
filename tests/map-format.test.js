const path = require('path');
const Generator = require('@asyncapi/generator');
const { readFile } = require('fs').promises;

const MAIN_TEST_RESULT_PATH = path.join('tests', 'temp', 'integrationTestResult');

const generateFolderName = () => {
    // you always want to generate to new directory to make sure test runs in clear environment
    return path.resolve(MAIN_TEST_RESULT_PATH, Date.now().toString());
};

describe('template integration tests for map format', () => {

    jest.setTimeout(30000);

    it('should generate DTO file with proper map types', async() => {
        const outputDir = generateFolderName();
        const params = {};
        const mapFormatExamplePath = './mocks/map-format.yml';

        const generator = new Generator(path.normalize('./'), outputDir, { forceWrite: true, templateParams: params });
        await generator.generateFromFile(path.resolve('tests', mapFormatExamplePath));

        const expectedFiles = [
            '/src/main/java/com/asyncapi/model/SongMetaData.java',
        ];
        for (const index in expectedFiles) {
            const file = await readFile(path.join(outputDir, expectedFiles[index]), 'utf8');
            expect(file).toMatchSnapshot();
        }
    });
});
