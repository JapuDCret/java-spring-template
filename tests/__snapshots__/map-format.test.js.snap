// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`template integration tests for map format should generate DTO file with proper map types 1`] = `
"package com.asyncapi.model;

import javax.validation.constraints.*;
import javax.validation.Valid;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonValue;

import java.util.List;
import java.util.Map;
import java.util.Objects;


public class SongMetaData {
    
    private @Valid Map<String, String> tags;
    
    private @Valid Map<String, Long> stats;
    
    private @Valid Map<String, Interpret> interprets;
    

    

    /**
     * Tags
     */
    @JsonProperty(\\"tags\\")
    public Object getTags() {
        return tags;
    }

    public void setTags(Object tags) {
        this.tags = tags;
    }
    

    /**
     * Stats
     */
    @JsonProperty(\\"stats\\")
    public Object getStats() {
        return stats;
    }

    public void setStats(Object stats) {
        this.stats = stats;
    }
    

    /**
     * Interprets
     */
    @JsonProperty(\\"interprets\\")
    public Map<String, Interpret> getInterprets() {
        return interprets;
    }

    public void setInterprets(Map<String, Interpret> interprets) {
        this.interprets = interprets;
    }
    
    @Override
    public boolean equals(Object o) {
        if (this == o) {
            return true;
        }
        if (o == null || getClass() != o.getClass()) {
            return false;
        }
        SongMetaData songMetaData = (SongMetaData) o;
        return 
            Objects.equals(this.tags, songMetaData.tags) &&
            Objects.equals(this.stats, songMetaData.stats) &&
            Objects.equals(this.interprets, songMetaData.interprets);
    }

    @Override
    public int hashCode() {
        return Objects.hash(tags, stats, interprets);
    }

    @Override
    public String toString() {
        return \\"class SongMetaData {\\\\n\\" +
        
                \\"    tags: \\" + toIndentedString(tags) + \\"\\\\n\\" +
                \\"    stats: \\" + toIndentedString(stats) + \\"\\\\n\\" +
                \\"    interprets: \\" + toIndentedString(interprets) + \\"\\\\n\\" +
                \\"}\\";
    }

    /**
     * Convert the given object to string with each line indented by 4 spaces (except the first line).
     */
    private String toIndentedString(Object o) {
        if (o == null) {
           return \\"null\\";
        }
        return o.toString().replace(\\"\\\\n\\", \\"\\\\n    \\");
    }
}"
`;
